### ISSUE TITLE
Fix bug where ' character breaks during template generation

### ISSUE THREAD

**Issue by silbinarywolf** [silbinarywolf/silverstripe-components#1]
This should be relatively easy to fix, I'll just make passing the HTML use a heredoc string.
I'll also add a test to catch this issue.

EDIT: Significant changes in the `dev` branch since this was posted. This might not even be a bug / issue anymore. Will need to add a test for to make sure quotes dont break things.

**Comment by paulkotzsymbiote** [comment]
This is probably related to an issue I'm having when passing single quotes in a text string eg:

<:NewsListItemSummary ArticleSummary="There once was a Gaul called Asterix whose friend Obelix carried menhirs all over the shop.  I don't know why he did that." />

Escaping the single quote eg: I don\'t know solves the problem.

Using stuff like .NoHTML() or LimitWordCountXML() don't make a difference.

**Comment by silbinarywolf** [comment]
Issue is that under the hood, it'd output PHP code like this:
```php
$val .= 'There once was a Gaul called Asterix whose friend Obelix carried menhirs all over the shop. I don't know why he did that.'
```

Because it's not escaping the ' with a \, it's making the string terminate early. Just need to make the parser terminate ' quotes.

**Comment by silbinarywolf** [comment]
Fixed in 0.3.1
